### PR TITLE
[APP-3853] Remove non-ASCII characters from sensor-reading errors

### PR DESCRIPTION
**Pull request by penguinland**
See comment in the code for the explanation, and see the ticket for how we tracked that down.

### CODE REVIEW

File: protoutils/sensor.go [hunk 1]
```diff
@@ -91,7 +94,18 @@ func goToProto(v interface{}) (*structpb.Value, error) {
...
-	return structpb.NewValue(v)
+	vv, err := structpb.NewValue(v)
+	// The structpb package sometimes inserts non-breaking spaces into their error messages (see
+	// https://github.com/protocolbuffers/protobuf-go/blob/master/internal/errors/errors.go#L30).
+	// However, we put error messages into the headers/trailers of GRPC responses, and client-side
+	// parsing of those requires that they must be entirely ASCII. So, we need to ensure that any
+	// errors from here do not contain non-breaking spaces.
+	if err != nil {
+		ascii := strings.ReplaceAll(
+			err.Error(), " " /* non-breaking space */, " " /* normal space */)
```
Comment (benjirewis) [inline protoutils/sensor.go:105]: How exactly does this replace non-breaking spaces? I guess `" "` and `" "` look exactly the same in this GH diff. Are they?

Comment (penguinland) [inline protoutils/sensor.go:105]: :raised_eyebrow: In Firefox, the two spaces appear indistinguishable. but I promise you that in vim, they're different.

Comment (penguinland) [reply protoutils/sensor.go:105]: Yeah, that's odd. In vim, they're different. In Firefox, they sure look the same. 
```
alan-17:09>cat sensor.go|head -n 105|tail -n 1 | hexdump
00000000  09 09 09 65 72 72 2e 45  72 72 6f 72 28 29 2c 20  |...err.Error(), |
00000010  22 c2 a0 22 20 2f 2a 20  6e 6f 6e 2d 62 72 65 61  |".." /* non-brea|
00000020  6b 69 6e 67 20 73 70 61  63 65 20 2a 2f 2c 20 22  |king space */, "|
00000030  20 22 20 2f 2a 20 6e 6f  72 6d 61 6c 20 73 70 61  | " /* normal spa|
00000040  63 65 20 2a 2f 29 0a                              |ce */).|
00000047
```
Usually a space is 0x20, but note how the second line near the beginning has 0xC2 0xA0. That's the non-breaking space.

Comment (benjirewis) [reply protoutils/sensor.go:105]: Cool thank you for the hexdump! I buy it.

**Review by benjirewis — APPROVED** [review]
LGTM! Thanks for looking into this.

**Pull request merged**